### DISCUSSION TITLE
Weird transparency of cells with same background color as Ghostty

### DISCUSSION THREAD

**Discussion by blt-r** [ghostty-org/ghostty#3573] [Feature Requests, Ideas] [answered]
Here's what I mean:

![image](https://github.com/user-attachments/assets/2bef45d9-baf4-4309-8f50-d58422fc1a38)

As you can see, when background opacity is <1, cells with exactly the same background color as the background color of the terminal are also transparent. But not as transparent as the actual background. And cells with even slightly different background color are fully opaque. 

This looks bad when I have the same theme in my terminal and editor.

I would expect either of these to happen:
- Cells with any background color are always as transparent as background.
- Cells with any background color are always opaque.

Steps to reproduce: Set `background-opacity = 0.5`. In fish shell, you can set background color to rgb value with `set_color -b 282C34` (#282C34 is background color of the default theme).

Environment: Fedora 41, GNOME Wayland, Ghostty built from main.

**Comment by timoclsn** [comment, marked as answer]
Oh, this behaviour would explain my config issue: https://github.com/ghostty-org/ghostty/discussions/3579

**Reply by timoclsn** [reply]
I now have worked around it by overriding the terminal background with a sliiiightly other background color so that it doesn't match my editor theme.

```
theme = tokyonight
# Slightly changed background color so that neovim background doesn't get transparent
background = 1A1B25 

```

**Reply by blt-r** [reply]
I started using different theme in the terminal. Just realized I don't really care about the terminal theme
